worksheet.rb

```diff
@@ -1,34 +1,175 @@
-########################################################
-# Step 1: Establish the layers
-
-# In this section of the file, as a series of comments,
-# create a list of the layers you identify.
-# Which layers are nested in each other?
-# Which layers of data "have" within it a different layer?
-# Which layers are "next" to each other?
-
-########################################################
-# Step 2: Assign a data structure to each layer
-
-# Copy your list from above, and in this section
-# determine what data structure each layer should have
-
-########################################################
-# Step 3: Make the data structure!
-
-# Setup the entire data structure:
-# based off of the notes you have above, create the
-# and manually write in data presented in rides.csv
-# You should be copying and pasting the literal data
-# into this data structure, such as "DR0004"
-# and "3rd Feb 2016" and "RD0022"
-
-########################################################
-# Step 4: Total Driver's Earnings and Number of Rides
-
-# Use an iteration blocks to print the following answers:
-# - the number of rides each driver has given
-# - the total amount of money each driver has made
-# - the average rating for each driver
-# - Which driver made the most money?
-# - Which driver has the highest average rating?
\ No newline at end of file
+# Find number of rides each driver has given
+def find_num_of_rides(rideshare)
+  num_of_rides = rideshare.map { |driver, rides| [driver, rides.length] }.to_h
+  return num_of_rides
```

Comment:
Assigning to `num_of_rides` wasn't necessary here (since all you do is return it).  You can save yourself a little work if you just directly return results.

```ruby
return rideshare.map { |driver, rides| [driver, rides.length] }.to_h
```